Source/WebCore/css/StyleRule.h

```diff
@@ -56,6 +56,8 @@ class StyleRuleBase : public RefCounted<StyleRuleBase> {
     WTF_MAKE_STRUCT_FAST_ALLOCATED_WITH_HEAP_IDENTIFIER(StyleRuleBase);
 public:
     StyleRuleType type() const { return static_cast<StyleRuleType>(m_type); }
+
+    virtual ~StyleRuleBase() = default;
```

Comment:
We should keep `StyleRuleBase` non virtual, and implement the trait system for it (cf the functions at the end of StyleRule.h) 

```
SPECIALIZE_TYPE_TRAITS_BEGIN(WebCore::StyleRulePositionTry)
    static bool isType(const WebCore::StyleRuleBase& rule) { return rule.isPositionTry(); }
SPECIALIZE_TYPE_TRAITS_END()
```

Reply:
I changed it back to the original.